src/python/pants/pantsd/pants_daemon.py

```diff
@@ -82,6 +82,17 @@ def buffer(self):
     return self
 
 
+class PantsDaemonSignalHandler(SignalHandler):
+
+  def __init__(self, services):
+    super(PantsDaemonSignalHandler, self).__init__()
+    self._services = services
+
+  def handle_sigint(self, signum, _frame):
+    for service in self._services:
```

Comment:
This should maybe terminate itself but not watchman (with [this function](https://github.com/pantsbuild/pants/blob/2a46298b3ad112244b42e2b8fe80719baf20b7c4/src/python/pants/pantsd/pants_daemon.py#L488)).

Something like:
```
class PantsDaemonSignalHandler(SignalHandler):
   def __init__(self, daemon):
    super(PantsDaemonSignalHandler, self).__init__()
    self._daemon = daemon

   def handle_sigint(self, signum, _frame):
    self._daemon.terminate(include_watchman=False)
    # Send a useful message to users
```